ofrak_patch_maker/ofrak_patch_maker/patch_maker.py

```diff
@@ -229,43 +238,85 @@ def make_bom(
-            if asm_file.endswith(".S"):
-                asm_file = self._toolchain.preprocess(asm_file, header_dirs, out_dir=out_dir)
-            assembled_object_path = self._toolchain.assemble(asm_file, header_dirs, out_dir=out_dir)
-            assembled_object = self.prepare_object(assembled_object_path)
-            object_map.update({original_asm_file: assembled_object})
+        # Compile .c files in parallel
+        c_files = list(filter(lambda x: x.endswith(".c"), source_list))
+        c_args = zip(
+            c_files,
+            itertools.repeat(header_dirs),
+            itertools.repeat(out_dir),
+            itertools.repeat(self._toolchain),
+            itertools.repeat(self.prepare_object),
+            itertools.repeat(self),
+            itertools.repeat(SourceFileType.C),
+        )
+        workers = math.ceil(cpu_count())
+        with get_context("spawn").Pool(processes=workers) as pool:
+            result = pool.starmap(
+                _create_object_file, c_args, chunksize=math.ceil(len(c_files) / workers)
+            )
+            # Close and join pool to give pytest-cov workers time to save coverage results
+            #  https://pytest-cov.readthedocs.io/en/latest/subprocess-support.html
+            pool.close()
+            pool.join()
+        for r in result:
+            object_map.update(r)
+
+        # Assemble .asm files in parallel
+        asm_files = list(filter(lambda x: x.endswith(".as") or x.endswith(".S"), source_list))
+        asm_args = zip(
+            asm_files,
+            itertools.repeat(header_dirs),
+            itertools.repeat(out_dir),
+            itertools.repeat(self._toolchain),
+            itertools.repeat(self.prepare_object),
+            itertools.repeat(self),
+            itertools.repeat(SourceFileType.ASM),
+        )
+        workers = math.ceil(cpu_count())
+        with get_context("spawn").Pool(processes=workers) as pool:
+            result = pool.starmap(
+                _create_object_file, asm_args, chunksize=math.ceil(len(asm_files) / workers)
+            )
+            # Close and join pool to give pytest-cov workers time to save coverage results
+            #  https://pytest-cov.readthedocs.io/en/latest/subprocess-support.html
+            pool.close()
+            pool.join()
+        for r in result:
+            object_map.update(r)
 
         # Compute the required size for the .bss segment
         bss_size_required = 0
-        symbols: Dict[str, int] = {}
+        symbols: Dict[str, Tuple[int, LinkableSymbolType]] = {}
+        unresolved_symbols: Set[str] = set()
         for o in object_map.values():
             bss_size_required += o.bss_size_required
             symbols.update(o.symbols)
+            # Resolve symbols defined within different patch files within the same patch BOM
+            for sym in o.rel_symbols.keys():
+                # Have not already seen this symbol in a previous patch object
```

Comment:
If the symbol is defined in a later object file, we'll add it to the set of unresolved symbols here and never remove it. This logic needs to be done either in two passes (get two separate sets of all `symbols` and of all `rel_symbols`, then subtract the aggregated defined `symbols` set from the aggregated undefined `rel_symbols` set) or if you want to build it as you go, it needs to remove a symbol from the `unresolved_symbols` set if it's in an object's `symbols` (NOT if it is in an object's `rel_symbols`, as the previous implementation of this code did).

I think we need a unit test for this logic.